### PR TITLE
[1645] Trainee course age range as two integers

### PR DESCRIPTION
**Pull request by jhenderson**
### Context
https://trello.com/c/QjV0PHVA/1645-m-age-range-as-two-integers

### Changes proposed in this pull request
- Add new attributes to `Trainee`: `course_min_age` and `course_max_age` to replace `age_range`
- Update `AgeRange` module to represent age ranges as arrays
- Add `data_migrate` gem to allow data migrations to happen as a separate rake task and decouple it from DB migrations
- Create new data migration to populate `course_min_age` and `course_max_age` from `Trainee#age_range` (`age_range` column will be removed in a separate PR)
- Add `data:migrate` to `Dockerfile`
- Minor refactoring in in some files that are part of the PR

### Guidance to review
- Create a new "School direct (salaried)" trainee and visit its course details edit page
- Check validation still works and then enter valid parameters
- Check the confirm page shows all the details you entered

### CODE REVIEW

**Comment by jhenderson** [comment]
[[M] Age range as two integers](https://trello.com/c/QjV0PHVA/1645-m-age-range-as-two-integers)

File: app/services/trainees/create_from_apply.rb [hunk 1]
```diff
@@ -43,7 +43,8 @@ def mapped_attributes
         email: raw_contact_details["email"],
         training_route: course&.route,
         subject: course&.name,
-        age_range: course&.age_range,
+        course_min_age: course&.min_age,
```
Comment (AbigailMcP) [inline app/services/trainees/create_from_apply.rb:46]: Is there a reason we're preferring to set the `course_min_age` and `course_max_age` in this service, but setting `course_age_range: course&.age_range` in the `ConfirmPublishCourseForm`?

**Review by AbigailMcP — APPROVED** [review]
Woop woop! Nice work 👌

File: spec/features/trainees/edit_course_details_spec.rb [hunk 2]
```diff
@@ -75,19 +71,22 @@ def when_i_visit_the_course_details_page
...
-      course_details_page.public_send("main_age_range_#{template.age_range.parameterize(separator: '_')}").choose
+      course_details_page.public_send("main_age_range_#{trainee.course_age_range.join('_to_')}").choose
     else
       course_details_page.main_age_range_other.choose
-      course_details_page.additional_age_range.select(template.age_range)
+      course_details_page.additional_age_range.select(trainee.course_age_range.join(" to "))
     end
   end
 
+  def and_i_enter_invalid_parameters; end
```
Comment (AbigailMcP) [inline spec/features/trainees/edit_course_details_spec.rb:88]: Empty method :(

Comment (jhenderson) [reply spec/features/trainees/edit_course_details_spec.rb:88]: Good spot - I was intending on fleshing it out, but didnt need to in the end.

File: app/services/trainees/create_from_apply.rb (hunk 1 above)
Comment (jhenderson) [reply app/services/trainees/create_from_apply.rb]: No reason - good spot. I've made them both consistent.

**Comment by GeoffJohnson** [comment]
All done

**Pull request merged**